### PR TITLE
storage/bucket: add label accessor methods

### LINKED ISSUE #2370: Add GCS bucket labels feature

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #2370

#### To Dos

- [x] Docs
- Tests
  - [x] Unit
  - [x] System

This adds three methods:

- `bucket.deleteLabels([labelName, ...], function(err) {})`
- `bucket.getLabels(function(err, labels) {})`
- `bucket.setLabels({ labelName: 'labelValue' }, function(err) {})`

### CODE REVIEW

**Comment by danoscarmike** [comment]
@jskeet for vis.

**Comment by jskeet** [comment]
Not sure what you mean by "for via" - I'm not really an appropriate person to review Node code.

**Comment by stephenplusplus** [comment]
Probably "for visibility" -- can you tag someone else?

**Comment by jskeet** [comment]
I'm afraid I don't know anyone offhand to review the Node code here... And I'm only on my phone, so can't really look up folks appropriately.

**Comment by stephenplusplus** [comment]
It's not so much the code that needs review from a GCS member, but the methods we've created to support this feature, and the ways they can be used. We want to be sure we are supporting the intended use cases properly.

Having someone read over the code comment blocks in the PR would be helpful. They will demonstrate the methods and list their possible arguments.

**Comment by jskeet** [comment]
I still don't think I'm the most appropriate person to do this, and I wouldn't be to do it before Monday anyway.

**Comment by jskeet** [comment]
(Note that I'm not part of GCS myself.)

**Comment by stephenplusplus** [comment]
Okay, sorry for getting you looped up in here. @danoscarmike had you listed as the POC for Storage Veneer.

@danoscarmike any suggestions on how to proceed?

**Comment by danoscarmike** [comment]
Hi both, apologies Jon - thought you'd be able to review the approach as
opposed to the code itself - I'll find somebody on GCS team to have a look.

File: packages/storage/src/bucket.js [hunk 1]
```diff
@@ -627,6 +627,72 @@ Bucket.prototype.deleteFiles = function(query, callback) {
...
+ *
+ * //-
+ * // Delete a single label.
+ * //-
+ * bucket.deleteLabels('label', function(err, metadata) {});
+ *
+ * //-
+ * // Delete a specific set of labels.
+ * //-
+ * bucket.deleteLabels(['label', 'labeltwo'], function(err, metadata) {});
```
Comment (swcloud) [inline packages/storage/src/bucket.js]: To be consistent with the unit test below, maybe we should use 

bucket.deleteLabels(['labelone', 'labeltwo']

File: packages/storage/src/bucket.js [hunk 2]
```diff
@@ -1032,6 +1135,41 @@ Bucket.prototype.makePublic = function(options, callback) {
...
+ * This makes an underlying call to {module:storage/bucket#setMetadata}, which
+ * is a PATCH request. This means an individual label can be overwritten, but
+ * unmentioned labels will not be touched.
+ *
+ * @param {type} labels - Labels to set on the bucket.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.metadata - The bucket's metadata.
+ *
+ * @example
```
Comment (swcloud) [inline packages/storage/src/bucket.js]: To be consistent with the unit test below, maybe we should use

var labels = {
   labelone: 'labelonevalue',
   labeltwo: 'labeltwovalue'
};

File: packages/storage/src/bucket.js (hunk 1 above)
Comment (stephenplusplus) [inline packages/storage/src/bucket.js]: Good catch, will do!

File: packages/storage/src/bucket.js (hunk 2 above)
Comment (stephenplusplus) [inline packages/storage/src/bucket.js]: Done!

**Pull request merged**